src/loader/init.c

```diff
@@ -0,0 +1,149 @@
+#include <postgres.h>
+#include <pg_config.h>
+#include <access/xact.h>
+#include <commands/extension.h>
+#include <miscadmin.h>
+#include <utils/guc.h>
+#include <utils/inval.h>
+#include <parser/analyze.h>
+#include <nodes/print.h>
+
+
+#include "extension.h"
+
+/*
+ * Some notes on design:
+ *
+ * We do not check for the installation of the extension upon loading the extension and instead rely on a hook for two reasons:
+ * 1) We probably can't
+ *	- The shared_preload_libraries is called in PostmasterMain which is way before InitPostgres is called.
+ *			(Note: This happens even before the fork of the backend)
+ *	-- This means we cannot query for the existance of the extension yet because the caches are initialized in InitPostgres.
+ * 2) We actually don't want to load the extension in two cases:
+ *	  a) We are upgrading the extension.
+ *	  b) We set the guc timescaledb.disable_load.
+ *
+ *
+ */
+
+#define MIN_SUPPORTED_VERSION_STR "9.6"
```

Comment:
I think we should probably remove this now that we have a check in CMake for version. If we want to keep something like this, we should generate a header file in CMake containing this stuff.